### ISSUE TITLE
Environment injected in script instead of in shell

### ISSUE THREAD

**Issue by Bigpet** [nektos/act#407]
I've run into an incompatibility with the GitHub Actions runner. When setting a custom shell command act seems to assume a normal shell whereas the GitHub Actions runner can deal with things like `python` just fine.  

```
name: Test act
on:
  release:
jobs:
  test_shell_run:
    name: Test shell run
    runs-on: ubuntu-latest
    steps:
      - name: Test script prepend
        id: full_output
        shell: cat {0}
        run: |
          some other non bash scripting lang
          is written here
```

This creates a shell script `full_output`, but doesn't just dump the "run:" commands in it, but also inserts some shell environment variable commands into the file (namely "PATH=":${PATH}"")

It would be great if the environment stuff would be setup into the environment that "shell" is executed in, instead of trying to set it up from inside.

### expected output
```
[Test act/Test shell run]   🐳  docker run image=node:12.6-buster-slim entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test act/Test shell run] ⭐  Run Test script prepend
| some other non bash scripting lang
| is written here
[Test act/Test shell run]   ✅  Success - Test script prepend
```
### actual output
```
[Test act/Test shell run]   🐳  docker run image=node:12.6-buster-slim entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test act/Test shell run] ⭐  Run Test script prepend
| PATH=":${PATH}"
| some other non bash scripting lang
| is written here
[Test act/Test shell run]   ✅  Success - Test script prepend
```

### version

act version 0.2.16
MacOs

**Comment by winksaville** [comment]
This should be fixed by change #406. If you checkout the master branch and execute `make build` and then execute  `dist/local/act` it should work. It worked for me:
```
$ ~/prgs/nektos/forks/act/dist/local/act 
[Test act/Test shell run] 🚀  Start image=ubuntu-with-hello
[Test act/Test shell run]   🐳  docker run image=ubuntu-with-hello entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test act/Test shell run] ⭐  Run Test script prepend
| some other non bash scripting lang
| is written here
[Test act/Test shell run]   ✅  Success - Test script prepend
```

**Comment by Bigpet** [comment]
Yep, that fixed my issue.

Thanks for the quick fix.
